src/app/core/campaign/campaign.service.ts

```diff
@@ -41,16 +33,16 @@ export class CampaignService {
     );
   }
 
-  updateCampaign(campaign: Campaign): Observable<HalDoc> {
-    return this.campaignDoc$.pipe(switchMap(doc => doc.update(campaign)));
+  updateCampaign(doc: HalDoc, campaign: Campaign): Observable<HalDoc> {
+    return doc.update(campaign);
   }
 
   createCampaign(campaign: Campaign): Observable<HalDoc> {
-    return this.augury.root.pipe(switchMap(rootDoc => rootDoc.create('prx:campaign', {}, campaign)));
+    return this.augury.root.pipe(mergeMap(rootDoc => rootDoc.create('prx:campaign', {}, campaign)));
   }
 
-  createFlight(flight: Flight): Observable<HalDoc> {
-    return this.campaignDoc$.pipe(switchMap(doc => doc.create('prx:flights', {}, flight)));
```

Comment:
Providing the campaign HalDoc as a parameter instead of hooking up to the state Observable fixes a nasty bug where new flights on new campaigns currently cannot be saved. The HalDoc observable wasn't firing until it saw a value from the persisted campaign, so the flight POST events weren't happening until some time later if you loaded a different campaign and they were posting to the wrong campaign. Gross. Didnt like that state tied up in here anyway.

Reply:
Yuck - good catch!